### PR TITLE
Minor: Add an example for backtrace pretty print

### PR DESCRIPTION
**Pull request by goldmedal**
## Which issue does this PR close?

No exact issue.

## Rationale for this change
Printing the backtrace directly isn't user-friendly for identifying the root cause. Here is an example to show how to print the backtrace in a way that is user- or IDE-friendly.

## What changes are included in this PR?
add an example in the doc.

## Are these changes tested?
no 

## Are there any user-facing changes?
no

### CODE REVIEW

File: docs/source/user-guide/crate-configuration.md [hunk 1]
```diff
@@ -141,6 +141,48 @@ To obtain a backtrace:
...
 cargo build --features=backtrace
 RUST_BACKTRACE=1 cargo test --features=backtrace --package datafusion --lib -- physical_planner::tests::test_get_backtrace_for_failed_code --exact --nocapture
+
+running 1 test
+Error: Plan("Invalid function 'row_numer'.\nDid you mean 'ROW_NUMBER'?\n\nbacktrace:    0: std::backtrace_rs::backtrace::libunwind::trace\n             at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/std/src/../../backtrace/src/backtrace/libunwind.rs:105:5\n   1: std::backtrace_rs::backtrace::trace_unsynchronized\n...
 ```
 
 Note: The backtrace wrapped into systems calls, so some steps on top of the backtrace can be ignored
+
+If you want to show the backtrace in a pretty-printed format, you can print the `DataFusionError` using `eprintln!("{e}");`.
```
Comment (comphead) [inline docs/source/user-guide/crate-configuration.md:151]:
```suggestion
To show the backtrace in a pretty-printed format use `eprintln!("{e}");`.
```

**Review by comphead — APPROVED** [review]
that looks good to me thanks @goldmedal

**Comment by alamb** [comment]
🚀

**Comment by goldmedal** [comment]
Thanks @comphead @alamb !

**Pull request merged**